### PR TITLE
Scripts/SQL: Mining

### PR DESCRIPTION
**Pull request by dazusu**

### CODE REVIEW

File: scripts/globals/mining.lua [hunk 1]
```diff
@@ -1,149 +1,497 @@
...
+        copy[k] = v
     end
 
-    return item;
+    local res = {}
+    while next(copy) do
+        res[#res + 1] = table.remove(copy, math.random(#copy))
+    end
+    return res
 end
```
Comment (xdemolish) [inline scripts/globals/mining.lua]: could you move this function to scripts/globals/common.lua
would be useful to have this for other stuff too

File: sql/npc_list.sql [hunk 2]
```diff
@@ -14484,12 +14484,13 @@ INSERT INTO `npc_list` VALUES (17363356,'Moogle','Moogle',0,0.000,0.000,0.000,0,
...
+INSERT INTO `npc_list` VALUES (17363362,'Mining_Point','Mining Point',0,186.116,-15.432,6.930,32775,40,40,0,0,0,0,3,0x0000780900000000000000000000000000000000,0,null);
+INSERT INTO `npc_list` VALUES (17363363,'Mining_Point','Mining Point',0,152.190,-16.394,142.365,32775,40,40,0,0,0,0,3,0x0000780900000000000000000000000000000000,0,null);
+INSERT INTO `npc_list` VALUES (17363364,'Mining_Point','Mining Point',0,290.244,-15.718,104.856,32774,40,40,0,0,0,0,3,0x0000780900000000000000000000000000000000,0,null);
+INSERT INTO `npc_list` VALUES (17363365,'Mining_Point','Mining Point',0,266.499,-32.401,-71.379,32774,40,40,0,0,0,0,3,0x0000780900000000000000000000000000000000,0,null);
+INSERT INTO `npc_list` VALUES (17363366,'Mining_Point','Mining Point',0,104.773,-32.418,82.766,32775,40,40,0,0,0,0,3,0x0000780900000000000000000000000000000000,0,null);
+INSERT INTO `npc_list` VALUES (17363367,'Treasure_Chest','Treasure Chest',227,250.037,-32.069,174.156,6,40,40,0,0,0,0,3,0x0000C00300000000000000000000000000000000,0,null);
 INSERT INTO `npc_list` VALUES (17363367,'Treasure_Chest','Treasure Chest',227,250.037,-32.069,174.156,6,40,40,0,0,0,0,3,0x0000C00300000000000000000000000000000000,0,null);
```
Comment (TeoTwawki) [inline sql/npc_list.sql]: Accidentally duplicated the treasure chest.

File: scripts/globals/mining.lua (hunk 1 above)
Comment (bendangelo) [inline scripts/globals/mining.lua]: Maybe just put it on the utils namespace `utils.lal`

Comment (TeoTwawki) [inline scripts/globals/mining.lua]: wtf is `lal` ? >.>

**Comment by dazusu** [comment]
Need to do more work on this thanks to some issues Teo made me aware of.

**Comment by xdemolish** [comment]
don't need to close, the pull request auto updates on pushing to the same branch

**Comment by dazusu** [comment]
All issues raised are now resolved.

**Comment by xdemolish** [comment]
@teschnei

**Pull request merged**